support-lib/jni/djinni_support.cpp

```diff
@@ -28,12 +28,39 @@ namespace djinni {
 // Set only once from JNI_OnLoad before any other JNI calls, so no lock needed.
 static JavaVM * g_cachedJVM;
 
+/*static*/
+JniClassInitializer::registration_vec & JniClassInitializer::get_vec() {
+    static JniClassInitializer::registration_vec m;
+    return m;
+}
+
+/*static*/
+std::mutex & JniClassInitializer::get_mutex() {
+    static std::mutex mtx;
+    return mtx;
+}
+
+JniClassInitializer::registration_vec JniClassInitializer::get_all() {
+    const std::lock_guard<std::mutex> lock(get_mutex());
+    return get_vec();
+}
+
+JniClassInitializer::JniClassInitializer(const std::function<void()> & init) : init(init){
+    const std::lock_guard<std::mutex> lock(get_mutex());
+    get_vec().emplace_back(this);
+}
+
+JniClassInitializer::~JniClassInitializer() {
+    const std::lock_guard<std::mutex> lock(get_mutex());
+    get_vec().erase(std::remove(get_vec().begin(), get_vec().end(), this), get_vec().end());
+}
+
 void jniInit(JavaVM * jvm) {
     g_cachedJVM = jvm;
 
     try {
-        for (const auto & kv : JniClassInitializer::Registration::get_all()) {
-            kv.second->init();
+        for (const auto & initializer : JniClassInitializer::get_all()) {
+            initializer->init();
```

Comment:
Sure - I just pushed a change. 
Let me know if that covers what you had in mind.